File: algorithmic-toolbox/src/week1/scala/GreatestCommonDivisor.scala
import scala.io.StdIn._

object GreatestCommonDivisor {

  def main(args: Array[String]) = {
    val in = new java.util.Scanner(System.in)
    println(gcd(in.nextLong, in.nextLong))
  }

  def gcd(a: Long, b: Long): Long = if (b == 0) a else gcd(b, a % b);
}

File: algorithmic-toolbox/src/week1/java/GreatestCommonDivisor.java
package week1.java;
import java.util.Scanner;

class GreatestCommonDivisor {

    public static void main(final String[] args) {
        try (final Scanner in = new Scanner(System.in)) {
            System.out.println(gcd(in.nextLong(), in.nextLong()));
        }
    }

    private static long gcd(final long a, final long b) {
        return (b == 0) ? a : gcd(b, a % b);
    }
}
